### PR TITLE
Azure Monitor: Basic Logs modal acknowledgement

### PR DESCRIPTION
**Pull request by aangelisc**
**What is this feature?**

This PR adds functionality of a modal so that users always have to acknowledge that they want to switch from Analytics to Basic Logs:

image

this ensures that Editors are aware of the per - query costs before switching to basic

If the modal is not acknowledged, they will remain in Analytics mode.

Fixes #

#84949 

Manual copy of #86244 for CI

### CODE REVIEW

File: public/app/plugins/datasource/azuremonitor/components/LogsQueryEditor/LogsManagement.tsx [hunk 1]
```diff
@@ -1,31 +1,52 @@
...
 
 export function LogsManagement({ query, onQueryChange: onChange }: AzureQueryEditorFieldProps) {
+  const [basicLogsAckOpen, setBasicLogsAckOpen] = useState<boolean>(false);
   return (
     <>
+      <ConfirmModal
+        isOpen={basicLogsAckOpen}
+        title="Basic Logs Queries"
+        body="Are you sure you want to switch to Basic Logs?"
+        description="Basic Logs queries incur cost based on the amount of data scanned. Please acknowledge this panel is subject to per-query costs."
```
Comment (bossinc) [inline public/app/plugins/datasource/azuremonitor/components/LogsQueryEditor/LogsManagement.tsx:17]:
```suggestion
        description="Basic Logs queries incur cost based on the amount of data scanned."
```
The second sentence is redundant.

File: public/app/plugins/datasource/azuremonitor/components/LogsQueryEditor/LogsManagement.tsx [hunk 2]
```diff
@@ -1,31 +1,52 @@
...
-          onChange={(val) => {
-            let updatedBasicLogsQuery = setBasicLogsQuery(query, val);
-            if (val) {
-              // if basic logs selected, set dashboard time
-              updatedBasicLogsQuery = setDashboardTime(updatedBasicLogsQuery, 'dashboard');
-            }
-            onChange(setKustoQuery(updatedBasicLogsQuery, ''));
-          }}
-        />
+        <>
```
Comment (bossinc) [inline public/app/plugins/datasource/azuremonitor/components/LogsQueryEditor/LogsManagement.tsx:33]: Are the empty tags needed?

**Review by bossinc — APPROVED** [review]

**Comment by bossinc** [comment]
Also is this the correct branch or should it be main?

File: public/app/plugins/datasource/azuremonitor/components/LogsQueryEditor/LogsManagement.tsx (hunk 1 above)
Comment (jcolladokuri) [reply public/app/plugins/datasource/azuremonitor/components/LogsQueryEditor/LogsManagement.tsx:17]: Will be addressing this and the bottom comment in a separate PR :)

**Pull request closed without merging**